### PR TITLE
SandboxNextFixture replaced by SOX-based SandboxFixture

### PR DESCRIPTION
**Pull request by tudor-da**
This PR:
* replaces all usages of `SandboxNextFixture` by `SandboxFixture` backed by Sandbox-on-X.
* replaces all direct usages of sandbox-next server with a Sandbox-on-X -backed `SandboxServer`

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-assistant/daml-helper/BUILD.bazel [hunk 1]
```diff
@@ -94,7 +94,7 @@ da_haskell_test(
     srcs = ["test/DA/Daml/Helper/Test/Tls.hs"],
     data = [
         "daml-helper",
-        "//ledger/sandbox:sandbox-binary",
+        "//ledger/sandbox-:sandbox-binary",
```
Comment (cocreature) [inline daml-assistant/daml-helper/BUILD.bazel:97]:
```suggestion
        "//ledger/sandbox-classic:sandbox-binary",
```
?

Comment (tudor-da) [reply daml-assistant/daml-helper/BUILD.bazel:97]: Hmm, I tried to, but more things needed changing. This is me backtracking.

Comment (tudor-da) [reply daml-assistant/daml-helper/BUILD.bazel]: Reverted for the moment

File: ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/LedgerBridge.scala [hunk 2]
```diff
@@ -128,7 +128,7 @@ object LedgerBridge {
     val party = s.hint.getOrElse(UUID.randomUUID().toString)
     Update.PartyAddedToParticipant(
       party = Ref.Party.assertFromString(party),
-      displayName = s.displayName.getOrElse(party),
+      displayName = s.displayName.getOrElse(""),
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/LedgerBridge.scala:131]: o..O well spotted!

**Review by mziolekda — APPROVED** [review]
To me those changes look ok, I can approve the participant bits, but I would like the code-owners of the affected projects to give their approval i.e. `triggers`, `daml-script`, `ledger-service`, `language-support`

**Review by mziolekda — COMMENTED** [review]
To me those changes look ok, I can approve the participant bits, but I would like the code-owners of the affected projects to give their approval i.e. `triggers`, `daml-script`, `ledger-service`, `language-support`

File: triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractFuncTests.scala [hunk 3]
```diff
@@ -319,7 +319,7 @@ abstract class AbstractFuncTests
         for {
           client <- ledgerClient(
             // Sufficiently low that the transaction is larger than the max inbound message size
-            maxInboundMessageSize = 300
+            maxInboundMessageSize = 100
```
Comment (cocreature) [inline triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractFuncTests.scala:322]: Why did this change?

File: ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/LedgerBridge.scala (hunk 2 above)
Comment (cocreature) [reply ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/LedgerBridge.scala:132]: Seems like something that we might want to have a test for somewhere?

**Review by cocreature — APPROVED** [review]
Thanks!

File: triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractFuncTests.scala (hunk 3 above)
Comment (tudor-da) [reply triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractFuncTests.scala:322]: It was not failing for 300 for some Maybe the behavior with SoX produces lower chunks?

Comment (cocreature) [reply triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractFuncTests.scala:322]: hm don’t really understand why but no issue with changing it.

**Comment by tudor-da** [comment]
> Seems like something that we might want to have a test for somewhere?

Good point. 4aadacd84d3f5739d7fd1d90702292838a6ae18d

File: ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/LedgerBridge.scala (hunk 2 above)
Comment (mziolekda) [reply ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/LedgerBridge.scala:132]: Test added

File: daml-assistant/daml-helper/BUILD.bazel (hunk 1 above)
Comment (tudor-da) [reply daml-assistant/daml-helper/BUILD.bazel]: Sandbox-next binaries not replaced in this PR.

**Comment by tudor-da** [comment]
@ray-roestenburg-da @stefanobaghino-da Can you have a look? The changes are quite trivial but they touch your domain so I'd like an approve from your team as well

File: daml-assistant/daml-helper/BUILD.bazel (hunk 1 above)
Comment (cocreature) [reply daml-assistant/daml-helper/BUILD.bazel]: doing that in https://github.com/digital-asset/daml/pull/12761

**Review by stefanobaghino-da — APPROVED** [review]
@tudor-da Thanks! 🙇🏻‍♂️ Are there other outstanding usages of `sandbox-next` or `sandbox-classic` that we should take care of? If yes, I'll make sure we keep track of that.

@ray-roestenburg-da Context for you: "sandbox" is the generic term by which we refer to implementations of the Ledger API that are mostly well suited for local development environment. Because of this, they are also well suited to be used for integration testing. Due to historical contingencies we ended up with three implementations of the sandbox:
- `sandbox-classic` is the earliest one, with one in-memory only implementations which has been extremely resilient in integration testing due to its speed and a persistent implementation that served as the basis for the Daml Driver for PostgreSQL -- the latter is a fully functional, persistent implementation of the Ledger API (albeit centralized) that however fell short of implementing the full architecture where the participant (the Ledger API server) was separated from the committer (the component that takes care of persisting transaction in a distributed environment) -- this architecture I described is usually referred to as "Daml-on-X" and was the first architecture aimed at providing Daml support on generic underlying storage mechanisms
- `sandbox-next` is the follow up that implemented the full architecture (briefly) described above -- it replaced in full `sandbox-classic` as the sandbox implementation provided alongside the SDK, but fell short of replacing `sandbox-classic` in our integration tests due to higher overhead
- `sandbox-on-x` is the "third generation", trying to close the loop and provide a full implementation of the "Daml-on-X" architecture while also addressing the performance issues of `sandbox-next`; note that while previous implementation were also aimed at SDK users, this is designed exclusively as an integration testing facility, with Canton running in single-node mode becoming the sandbox for SDK users

The plan is to kill `sandbox-classic` and `sandbox-next`, leaving only `sandbox-on-x` alive and lowering the pressure on the participant team to maintain several sandbox implementations. This PR is a step in that direction with regards to our integration testing infrastructure.

**Comment by tudor-da** [comment]
Thanks for your input @stefanobaghino-da . Wrt what remains to be done, there are some usages of the sandbox libraries that need to be switched over to sandbox-on-x:

`//ledger/sandbox-classic:sandbox-classic-binary` is used in 
```
//language-support/ts/codegen/tests:build-and-lint-test
//language-support/scala/codegen-sample-app:ScalaCodeGenIT
```

`//ledger/sandbox-classic:sandbox-classic` is used in 
```
//ledger/sandbox-perf:sandbox-perf
//ledger/sandbox-perf:sandbox-perf_compiled_benchmark_lib
//ledger/sandbox-perf:sandbox-perf_codegen_resources.jar
//ledger/sandbox-perf:sandbox-perf_codegen.srcjar
//ledger/sandbox-perf:sandbox-perf_codegen
//ledger/sandbox-perf:sandbox-perf_generator
//daml-script/export/integration-tests/matches-docs:example-export-compare
//daml-script/export/integration-tests/matches-docs:example-export-compare.sh
//daml-script/export/integration-tests/matches-docs:example-export-compare_script
//daml-script/export/integration-tests/matches-docs:example-export/daml.yaml
//daml-script/export/integration-tests/matches-docs:example-export/args.json
//daml-script/export/integration-tests/matches-docs:example-export/Export.daml
//daml-script/export/integration-tests/matches-docs:example-export
```

@mziolekda What should we do with `//ledger/sandbox-perf` ?

`//ledger/sandbox:sandbox` is used in 
```
//language-support/hs/bindings:test@hlint
//language-support/hs/bindings:test@ghci
//language-support/hs/bindings:test-repl
//language-support/hs/bindings:test@repl
//language-support/hs/bindings:repl
//daml-assistant/daml-helper:test-tls@hlint
//daml-assistant/daml-helper:test-tls@ghci
//daml-assistant/daml-helper:test-tls-repl
//daml-assistant/daml-helper:test-tls@repl
//daml-assistant/daml-helper:test-deployment@hlint
//daml-assistant/daml-helper:test-deployment@ghci
//daml-assistant/daml-helper:test-deployment-repl
//daml-assistant/daml-helper:test-deployment@repl
//daml-assistant/daml-helper:test-daml-packages@hlint
//daml-assistant/daml-helper:test-daml-packages@ghci
//daml-assistant/daml-helper:test-daml-packages-repl
//daml-assistant/daml-helper:test-daml-packages@repl
//daml-assistant/daml-helper:test-daml-packages
//daml-assistant/daml-helper:test-daml-ledger@hlint
//daml-assistant/daml-helper:test-daml-ledger@ghci
//daml-assistant/daml-helper:test-daml-ledger-repl
//daml-assistant/daml-helper:test-daml-ledger@repl
//daml-assistant/daml-helper:test-daml-ledger
//compiler/damlc/tests:repl@hlint
//compiler/damlc/tests:repl@ghci
//compiler/damlc/tests:repl-repl
//compiler/damlc/tests:repl@repl
//compiler/damlc/tests:repl-functests@hlint
//compiler/damlc/tests:repl-functests@ghci
//compiler/damlc/tests:repl-functests-repl
//compiler/damlc/tests:repl-functests@repl
//compiler/damlc/tests:repl-functests
//compiler/damlc/tests:repl
//compiler/damlc/tests:package-manager@hlint
//compiler/damlc/tests:package-manager@ghci
//compiler/damlc/tests:package-manager-repl
//compiler/damlc/tests:package-manager@repl
//compiler/damlc/tests:package-manager
//:repl
//language-support/hs/bindings:test
//daml-assistant/daml-helper:test-tls
//daml-assistant/daml-helper:test-deployment
```

**Comment by stefanobaghino-da** [comment]
> Thanks for your input @stefanobaghino-da . Wrt what remains to be done, there are some usages of the sandbox libraries that need to be switched over to sandbox-on-x:
> 
> `//ledger/sandbox-classic:sandbox-classic-binary` is used in
> 
> ```
> //language-support/ts/codegen/tests:build-and-lint-test
> //language-support/scala/codegen-sample-app:ScalaCodeGenIT
> ```
> 
> `//ledger/sandbox-classic:sandbox-classic` is used in
> 
> ```
> //ledger/sandbox-perf:sandbox-perf
> //ledger/sandbox-perf:sandbox-perf_compiled_benchmark_lib
> //ledger/sandbox-perf:sandbox-perf_codegen_resources.jar
> //ledger/sandbox-perf:sandbox-perf_codegen.srcjar
> //ledger/sandbox-perf:sandbox-perf_codegen
> //ledger/sandbox-perf:sandbox-perf_generator
> //daml-script/export/integration-tests/matches-docs:example-export-compare
> //daml-script/export/integration-tests/matches-docs:example-export-compare.sh
> //daml-script/export/integration-tests/matches-docs:example-export-compare_script
> //daml-script/export/integration-tests/matches-docs:example-export/daml.yaml
> //daml-script/export/integration-tests/matches-docs:example-export/args.json
> //daml-script/export/integration-tests/matches-docs:example-export/Export.daml
> //daml-script/export/integration-tests/matches-docs:example-export
> ```
> 
> @mziolekda What should we do with `//ledger/sandbox-perf` ?
> 
> `//ledger/sandbox:sandbox` is used in
> 
> ```
> //language-support/hs/bindings:test@hlint
> //language-support/hs/bindings:test@ghci
> //language-support/hs/bindings:test-repl
> //language-support/hs/bindings:test@repl
> //language-support/hs/bindings:repl
> //daml-assistant/daml-helper:test-tls@hlint
> //daml-assistant/daml-helper:test-tls@ghci
> //daml-assistant/daml-helper:test-tls-repl
> //daml-assistant/daml-helper:test-tls@repl
> //daml-assistant/daml-helper:test-deployment@hlint
> //daml-assistant/daml-helper:test-deployment@ghci
> //daml-assistant/daml-helper:test-deployment-repl
> //daml-assistant/daml-helper:test-deployment@repl
> //daml-assistant/daml-helper:test-daml-packages@hlint
> //daml-assistant/daml-helper:test-daml-packages@ghci
> //daml-assistant/daml-helper:test-daml-packages-repl
> //daml-assistant/daml-helper:test-daml-packages@repl
> //daml-assistant/daml-helper:test-daml-packages
> //daml-assistant/daml-helper:test-daml-ledger@hlint
> //daml-assistant/daml-helper:test-daml-ledger@ghci
> //daml-assistant/daml-helper:test-daml-ledger-repl
> //daml-assistant/daml-helper:test-daml-ledger@repl
> //daml-assistant/daml-helper:test-daml-ledger
> //compiler/damlc/tests:repl@hlint
> //compiler/damlc/tests:repl@ghci
> //compiler/damlc/tests:repl-repl
> //compiler/damlc/tests:repl@repl
> //compiler/damlc/tests:repl-functests@hlint
> //compiler/damlc/tests:repl-functests@ghci
> //compiler/damlc/tests:repl-functests-repl
> //compiler/damlc/tests:repl-functests@repl
> //compiler/damlc/tests:repl-functests
> //compiler/damlc/tests:repl
> //compiler/damlc/tests:package-manager@hlint
> //compiler/damlc/tests:package-manager@ghci
> //compiler/damlc/tests:package-manager-repl
> //compiler/damlc/tests:package-manager@repl
> //compiler/damlc/tests:package-manager
> //:repl
> //language-support/hs/bindings:test
> //daml-assistant/daml-helper:test-tls
> //daml-assistant/daml-helper:test-deployment
> ```
> 
> `//ledger/sandbox:sandbox-binary` is used in
> 
> ```
> //language-support/hs/bindings:test@hlint
> //language-support/hs/bindings:test@ghci
> //language-support/hs/bindings:test-repl
> //language-support/hs/bindings:test@repl
> //language-support/hs/bindings:repl
> //daml-assistant/daml-helper:test-tls@hlint
> //daml-assistant/daml-helper:test-tls@ghci
> //daml-assistant/daml-helper:test-tls-repl
> //daml-assistant/daml-helper:test-tls@repl
> //daml-assistant/daml-helper:test-deployment@hlint
> //daml-assistant/daml-helper:test-deployment@ghci
> //daml-assistant/daml-helper:test-deployment-repl
> //daml-assistant/daml-helper:test-deployment@repl
> //daml-assistant/daml-helper:test-daml-packages@hlint
> //daml-assistant/daml-helper:test-daml-packages@ghci
> //daml-assistant/daml-helper:test-daml-packages-repl
> //daml-assistant/daml-helper:test-daml-packages@repl
> //daml-assistant/daml-helper:test-daml-packages
> //daml-assistant/daml-helper:test-daml-ledger@hlint
> //daml-assistant/daml-helper:test-daml-ledger@ghci
> //daml-assistant/daml-helper:test-daml-ledger-repl
> //daml-assistant/daml-helper:test-daml-ledger@repl
> //daml-assistant/daml-helper:test-daml-ledger
> //compiler/damlc/tests:repl@hlint
> //compiler/damlc/tests:repl@ghci
> //compiler/damlc/tests:repl-repl
> //compiler/damlc/tests:repl@repl
> //compiler/damlc/tests:repl-functests@hlint
> //compiler/damlc/tests:repl-functests@ghci
> //compiler/damlc/tests:repl-functests-repl
> //compiler/damlc/tests:repl-functests@repl
> //compiler/damlc/tests:repl-functests
> //compiler/damlc/tests:repl
> //compiler/damlc/tests:package-manager@hlint
> //compiler/damlc/tests:package-manager@ghci
> //compiler/damlc/tests:package-manager-repl
> //compiler/damlc/tests:package-manager@repl
> //compiler/damlc/tests:package-manager
> //:repl
> //language-support/hs/bindings:test
> //daml-assistant/daml-helper:test-tls
> //daml-assistant/daml-helper:test-deployment
> ```
> 
> After removing these dependencies, we are fully reliant on Sandbox-on-X for SDK internal testing and allowing removal of `sandbox-classic` and `sandbox-next`. @moritzkiefer-da @ray-roestenburg-da @stefanobaghino-da Can you help with switching these over?

Thanks for the comprehensive summary of usages of older sandbox implementations. We can take care of the remaining usages I believe. We may have issues if there are instances in which `sandbox-on-x` falls short in terms of performance to the point where our tests end up failing, but we'll make sure to get in touch if anything like that arises.

@tudor-da @mziolekda Do you have a specific time frame in mind when it comes to phasing out older sandbox implementations?

**Pull request merged**